### PR TITLE
store/tikv:move error from kv into package error

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR moves `tikv/kv/error.go`  & "tikv/errno" into new package tikv/error.
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

**Review by andylokandy — APPROVED** [review]

File: store/tikv/error/errcode.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 PingCAP, Inc.
```
Comment (youjiali1995) [inline store/tikv/error/errcode.go:1]: Do you forget to remove `store/tikv/errno/errcode.go`?

**Review by youjiali1995 — APPROVED** [review]
LGTM

**Pull request merged**